Source/SwiftLintCore/Models/Version.swift

```diff
@@ -5,4 +5,34 @@ public struct Version {
 
     /// The current SwiftLint version.
     public static let current = Version(value: "0.55.1")
+
+    /// Public initializer.
+    ///
+    /// - parameter value: the string value for this version.
+    public init(value: String) {
+        self.value = value
+    }
+}
+
+extension Version: Comparable {
+    public static func == (lhs: Version, rhs: Version) -> Bool {
+        lhs.value == rhs.value
+    }
+
+    public static func < (lhs: Version, rhs: Version) -> Bool {
+        if let lhsComparators = lhs.comparators, let rhsComparators = rhs.comparators {
+            return lhsComparators.lexicographicallyPrecedes(rhsComparators)
+        }
+        return lhs.value < rhs.value
+    }
+
+    private var comparators: [Int]? {
+        let components = value.split(separator: ".").compactMap { Int($0) }
+        guard let major = components.first else {
+            return nil
+        }
+        let minor = components.dropFirst(1).first ?? 0
+        let patch = components.dropFirst(2).first ?? 0
+        return [major, minor, patch]
+    }
```

Comment:
There is a similar (or equal) implementation for `SwiftVersion`. Can we somehow reduce them to just one? Something like `VersionComparable` with a default implementation that only relies on the underlying version string.